src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Serialization/StringCache.cs

```diff
@@ -112,15 +109,16 @@ public bool Equals(Entry x, Entry y)
 
             public int GetHashCode(Entry obj)
             {
-                return obj.TargetHashCode;
+                return obj.GetHashCode();
             }
         }
 
+        // This can't be a struct because then Value Boxing would cause us to fail the ReferenceEquals check inside Equals.
```

Comment:
Dumb question: Instead of doing a ReferenceEquals on the Entry, could it do one on the WeakRef it holds?

Reply:
Not a dumb question, that's a good solution that my morning-brain didn't think of.